internal/collectors/contracts/erc20/approval_event.go

```diff
@@ -0,0 +1,117 @@
+package erc20
+
+import (
+	"context"
+	"log"
+	"math"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thepalbi/ethereum-prometheus-exporter/token"
+)
+
+type ApprovalEvent struct {
+	*Event
+}
+
+func NewERC20ApprovalEvent(client ContractClient, contractAddresses []common.Address, nowBlockNumber uint64) (*ApprovalEvent, error) {
+	clients := map[*contractInfo]*token.TokenFilterer{}
+	for _, contractAddress := range contractAddresses {
+		filterer, err := token.NewTokenFilterer(contractAddress, client)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create ERC20 approval evt collector")
+		}
+		info, err := getContractInfo(contractAddress, client)
+		if err != nil {
+			return nil, err
+		}
+
+		log.Printf("Got info for %s, symbol %s\n", info.Address, info.Symbol)
+		clients[info] = filterer
+	}
```

Comment:
This bit could be extracted as well right?

Reply:
Yes, easily. Done.